convert2rhel/unit_tests/utils_test.py

```diff
@@ -874,7 +888,298 @@ def communicate(self):
         pass
 
     def poll(self):
-        return 0
+        return self.return_code
+
+    def close(self):
+        pass
+
+
+@pytest.mark.parametrize(
```

Comment:
Also, @abadger, if you could take a good look over this, it would help so much! I feel that those unit tests could be done differently (in a better way), but I'm not sure how to improve it.